File: leetcode/lc-us/cpp/190.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res = 0;
        for (int i = 0; i < 32; i++) {
            auto digit = n & 1;
            res <<= 1;
            res |= digit;
            n >>= 1;
        }
        return res;
    }
};

File: leetcode/lc-us/rust/190.rs
impl Solution {
    pub fn reverse_bits(x: u32) -> u32 {
        let mut res: u32 = 0;
        for i in 0..32 {
            res <<= 1;
            if (x >> i) & 1 == 1 {
                res += 1;
            }
        }
        res
    }
}